nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java

```diff
@@ -124,7 +132,17 @@ private void waitForJobCompletion(String jobId, String query) throws IOException
     Instant deadline = Instant.now().plus(timeout);
 
     while (true) {
-      String responseBody = performHttpRequest(url, null);
+      String responseBody;
+      try {
+        responseBody = performHttpRequest(url, null);
+      } catch (IOException e) {
+        boolean rateLimited = e.getMessage() != null && e.getMessage().contains("http code: 429");
+        if (rateLimited) {
+          sleep(Duration.ofSeconds(60)); // ban duration
```

Comment:
Let's change this to 5 sec as well. One minute is quite long

Reply:
iiuc we are going to keep getting 429s for the next 60 seconds...
so if we sleep 5 sec here we are just looping again basically... but OK